dockertest/docker_daemon.py

```diff
@@ -159,16 +159,32 @@ def pid():
     if not stdout.startswith('MainPID='):
         raise RuntimeError("Unexpected output from %s: expected MainPID=NNN,"
                            " got '%s'" % (cmd_result.command, stdout))
-    return int(stdout[8:])
-
-
-def cmdline():
+    mainpid = int(stdout[8:])
+    cmd = cmdline(mainpid)
+    if 'dockerd' in cmd[0]:
+        return mainpid
+    # As of April 2017 we may be running dockerd under runc. If so, actual
+    # dockerd process is an immediate child of the one returned by systemd.
+    for child_pid in utils.run("pgrep -P %s" % mainpid).stdout.split():
+        cmd = cmdline(child_pid)
+        if 'dockerd' in cmd[0]:
+            return int(child_pid)
+    # Urp. No dockerd process found. Cross fingers & hope systemd is right.
+    # FIXME: we should warn, but we have no self with which to logwarning()
```

Comment:
Fix:  No ``self`` needed, you can just use ``logging.warning()`` directly in this context.  The only reason we have special logging methods on ``SubtestBase`` is to make the messages look pretty.